### DISCUSSION TITLE
Typescript module resolution for metro?

### DISCUSSION THREAD

**Discussion by KrastanD** [storybookjs/react-native#451] [Q&A] [answered]
I am importing a Button component into the `.storybook/stories/Button.stories.tsx` and typescript fails to resolve it. However if I import the same component from the same library in `.storybook/index.ts` it has no such issue. My understanding is that because index.ts doesn't directly import any of the files from the `stories/` folder, typescript is not able to find them. I think [this](https://storybook.js.org/docs/react/builders/webpack#typescript-module-resolution) would fix it if I was using webpack, is there something similar for metro? Or am I totally off base?

**Comment by dannyhw** [comment, marked as answer]
Could you provide some example code? Typescript should work fine, i would double check your typescript config is including the folder where you have stories.

**Reply by KrastanD** [reply]
Here is an example repo: https://github.com/KrastanD/example-react-native-ui. Thanks for your help!

**Reply by dannyhw** [reply]
@KrastanD  here you go :)
https://github.com/KrastanD/example-react-native-ui/pull/1

**Reply by dannyhw** [reply]
I just had to forcefully include the .storybook folder since I guess typescript was ignoring it because its "hidden" with `.`

**Reply by KrastanD** [reply]
Gotcha, that's good to know. Thanks for your help Danny! Congrats on the release, I've been loving the updates.
